psi4/src/psi4/libmints/basisset.cc

```diff
@@ -193,20 +196,35 @@ int BasisSet::n_frozen_core(const std::string &depth, SharedMolecule mol) {
             if (Z > 0) {
                 // Add ECPs to Z, the number of electrons less ECP-treated electrons
                 double ECP = n_ecp_core(mymol->label(A));
-                if (Z + ECP > 2) nfzc += 1;
-                if (Z + ECP > 10) nfzc += 4;
-                if (Z + ECP > 18) nfzc += 4;
-                if (Z + ECP > 36) nfzc += 9;
-                if (Z + ECP > 54) nfzc += 9;
-                if (Z + ECP > 86) nfzc += 16;
+                int delta = 0;
+                if (Z + ECP > 2)  delta = 2;
+                if (Z + ECP > 10) delta = 10;
+                if (Z + ECP > 18) delta = 18;
+                if (Z + ECP > 36) delta = 36;
+                if (Z + ECP > 54) delta = 54;
+                if (Z + ECP > 86) delta = 86;
                 if (Z + ECP > 108) {
```

Comment:
This set of if statements and duplicated numbers could be replaced with a ```std::vector<int>``` and a ```lower_bound``` call, like in #1452 .

Reply:
Neat, thanks!